packages/mobile/src/screens/track-screen/TrackScreenDetailsTile.tsx

```diff
@@ -200,9 +201,12 @@ export const TrackScreenDetailsTile = ({
   } = track
 
   const isOwner = owner_id === currentUserId
-  const hideFavorite = is_unlisted || (isPremiumContentEnabled && isPremium)
+  const hideFavorite =
+    is_unlisted || (isPremiumContentEnabled && !doesUserHaveAccess)
   const hideRepost =
-    is_unlisted || !isReachable || (isPremiumContentEnabled && isPremium)
+    is_unlisted ||
+    !isReachable ||
```

Comment:
nit: any reason reachable is in this one and not in favorite? also could pull out `const isInaccessiblePremiumTrack =  isPremiumContentEnabled && !doesUserHaveAccess` if so inclined